akka-actor/src/main/scala/akka/actor/dungeon/DeathWatch.scala

```diff
@@ -72,12 +85,25 @@ private[akka] trait DeathWatch { this: ActorCell ⇒
     watching.contains(subject) || (subject.path.uid != ActorCell.undefinedUid &&
       watching.contains(new UndefinedUidActorRef(subject)))
 
+  // TODO this should be removed and be replaced with `watching.get(subject)`
+  //   when all actor references have uid, i.e. actorFor is removed
+  private def watchingGet(subject: ActorRef): Option[Option[Any]] =
```

Comment:
OK, these comments suggest that all these special helpers can then be removed entirely. Might be good to do that (in another PR) rather than putting energy in wrapping the access